### PR TITLE
Add containerized build process.

### LINKED ISSUE #949: [local-volume] Add containerized build process
/area local-volume

To reduce dependency on local tools and libraries, we should support containerized builds.  Here's an [example](https://github.com/kubernetes-sigs/gcp-filestore-csi-driver/blob/master/Dockerfile)

### PR DESCRIPTION
**Pull request by cofyc**
fix #949

Old `build` target is renamed to `build-local`.

### CODE REVIEW

File: local-volume/provisioner/Makefile [hunk 1]
```diff
@@ -20,12 +20,26 @@ ifeq ($(VERSION),)
...
 IMAGE = $(REGISTRY)local-volume-provisioner:$(VERSION)
 MUTABLE_IMAGE = $(REGISTRY)local-volume-provisioner:latest
 
-all build:
+all: build
+.PHONY: all
+
+build:
+	docker run --rm -w /go/src/github.com/kubernetes-incubator/external-storage/local-volume/provisioner \
+		-v $(PWD)/../..:/go/src/github.com/kubernetes-incubator/external-storage -e CGO_ENABLED=0 -e GOOS=linux \
```
Comment (msau42) [inline local-volume/provisioner/Makefile]: Can we put these steps inside the Dockerfile?  Then all you have to do here is just run docker build.  This makes it easier to integrate into CI systems.

Comment (cofyc) [reply local-volume/provisioner/Makefile]: Then it's not possible to support `make quick-container`. How about we only support these targets?

- `make all/build`: same as before build on the host locally
- `make container`: build in the container and make a image
- `make quick-container`: removed
- `make push/clean/test`: same as before

Comment (cofyc) [reply local-volume/provisioner/Makefile]: Updated

**Pull request merged**